tests/test_meshtastic_utils_coverage.py

```
@@ -0,0 +1,926 @@
+
+        result = mu._get_device_metadata(client)
+
+        assert result["firmware_version"] == "unknown"
+        assert result["success"] is False
+
+    def test_get_device_metadata_raises_on_error(self):
+        """Test _get_device_metadata raises error when requested."""
+        client = Mock()
+        client.localNode = None
+
+        with pytest.raises(RuntimeError, match="no localNode.getMetadata"):
```

Ruff (0.15.4)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)